src/documentation/programming-guide.md

```diff
@@ -1079,13 +1052,185 @@ For blocking execution, append the `waitUntilFinish` method:
+
+{:.language-java}
+`CoderRegistry` contains a default mapping of coders to standard Java types for any Pipeline you create using the Beam SDK for Java. The following table shows the standard mapping:
+
+{:.language-java}
+| Java Type        | Default Coder         |
+|------------------|-----------------------|
+| Double           | DoubleCoder           |
+| Instant          | InstantCoder          |
+| Integer          | VarIntCoder           |
+| Iterable         | IterableCoder         |
+| KV               | KvCoder               |
+| List             | ListCoder             |
+| Map              | MapCoder              |
+| Long             | VarLongCoder          |
+| String           | StringUtf8Coder       |
+| TableRow         | TableRowJsonCoder     |
+| Void             | VoidCoder             |
+| byte[ ]          | ByteArrayCoder        |
+| TimestampedValue | TimestampedValueCoder |
+
+
+{:.language-py}
+`CoderRegistry` contains a default mapping of Coders to standard Python types for any Pipeline you create using the Beam SDK for Python. The following table shows the standard mapping:
+
+{:.language-py}
+| Python Type | Default Coder |
+|-------------|---------------|
+| int         | VarIntCoder   |
+| float       | FloatCoder    |
+| str         | BytesCoder    |
+| bytes       | StrUtf8Coder  |
+| Tuple       | TupleCoder    |
+
+
+
+##### Looking up a default coder
+
+{:.language-java}
+You can use the method CoderRegistry.getDefaultCoder to determine the default Coder for a Java type. You can access the CoderRegistry for a given Pipeline by using the method Pipeline.getCoderRegistry. This allows you to determine (or set) the default Coder for a Java type on a per-pipeline basis: i.e. "for this pipeline, verify that Integer values are encoded using BigEndianIntegerCoder."
+
+{:.language-py}
+You can use the method CoderRegistry.get_coder to determine the default Coder for a Python type. You can use `coders.registry` to access the CoderRegistry. This allows you to determine (or set) the default Coder for a Python type on a per-pipeline basis.
+
+##### Setting the default coder for a type
+
+{:.language-java}
+To set the default Coder for a Java type for a particular pipeline, you obtain and modify the pipeline's `CoderRegistry`. You use the method `Pipeline.getCoderRegistry` to get the `CoderRegistry` object, and then use the method `CoderRegistry.registerCoder` to register a new `Coder` for the target Java type.
+
+{:.language-java}
+The following example code demonstrates how to set a default Coder, in this case `BigEndianIntegerCoder`, for Integer values for a pipeline.
+
+{:.language-py}
+To set the default Coder for a Python type, you obtain and modify the `CoderRegistry`. You may use the `coders.registry` to get the `CoderRegistry` object, and then use the method `CoderRegistry.register_coder` to register a new Coder for the target Python type.
```

Comment:
Seems like another example of a paragraph that might be able to be written once with only the method names toggling?